train-system/src/main/java/hu/bme/mit/train/system/TrainSystem.java

```diff
@@ -31,21 +34,38 @@ public TrainUser getUser() {
 
 	public TrainTachograph getTachograph() { return tachograph;}
 
+	Thread periodicReferenceSpeedCheck = 	new Thread(() -> {
```

Comment:
This should be moved to the TrainController and be tested in the TrainSystemTest class.